src/ansiblelint/_prerun.py

```diff
@@ -162,10 +152,23 @@ def _install_galaxy_role() -> None:
     role_name = yaml['galaxy_info'].get('role_name', None)
     role_author = yaml['galaxy_info'].get('author', None)
     if not role_name:
-        role_name = os.path.dirname(".")
+        role_name = pathlib.Path(".").absolute().name
         role_name = re.sub(r'^{0}'.format(re.escape('ansible-role-')), '', role_name)
-    if not role_author or not re.match(r"[\w\d_]{2,}", role_name):
-        print(MISSING_GALAXY_NAMESPACE, file=sys.stderr)
+    fqrn = f"{role_author}.{role_name}"
+    if not re.match(r"[\w\d_]{2,}\.[a-z][a-z0-9_]+$", fqrn):
```

Comment:
```suggestion
    if not re.match(r"[a-z0-9][a-z0-9_]{2,}\.[a-z][a-z0-9_]+$", fqrn):
```
The document you link to (https://galaxy.ansible.com/docs/contributing/namespaces.html#galaxy-namespace-limitations) specifies explicit restrictions on the galaxy namespace that you are not really checking here.

Also it does not give any restrictions on the role name. Where do you have these from? I think there should also be a reference for them.

Reply:
Ouch, we only support one link per rule and the role requirements are listed at https://galaxy.ansible.com/docs/contributing/creating_role.html#role-names